components/tools/correlation.py

```
@@ -0,0 +1,81 @@
+from utils.utils import generate_correlation_plot, run_orion, summarize_result
+
+logger = logging.getLogger(__name__)
+
+
+@tool
+async def metrics_correlation(
+    metric1: Annotated[str, Field(description="First metric to analyze")] = "podReadyLatency_P99",
+    metric2: Annotated[str, Field(description="Second metric to analyze")] = "ovnCPU_avg",
+    *,
+    config_name: ConfigParam = None,
+    since: Annotated[str, Field(description="Date to begin looking back for performance data")] = None,
```

Ruff (0.16.3)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)